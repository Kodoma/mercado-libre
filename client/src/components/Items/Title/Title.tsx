import React from "react";
import styles from "../Items.module.css";
import {Link, Typography} from "@material-ui/core";

const Title: React.FunctionComponent<TitleProps> = ({id, title}) => {
    return (
        <Link href={`/items/${id}`} className={styles.title}>
            <Typography variant="body2" gutterBottom>
                { title }
            </Typography>
        </Link>
    );
}

export default Title;

interface TitleProps {
    id: string;
    title: string;
}