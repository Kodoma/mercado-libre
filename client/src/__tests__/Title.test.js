import React from 'react';
import { shallow } from 'enzyme';
import Title from "../components/Items/Title/Title";

describe('tests Title', () => {
    it('should render my Title component', () => {
        const component = shallow(<Title id={1} title={'iphone'}/>);
        expect(component.getElements()).toMatchSnapshot();
    });
});