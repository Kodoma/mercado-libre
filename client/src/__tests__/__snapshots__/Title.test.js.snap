// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__tests__ Title should render my Title component 1`] = `
Array [
  <WithStyles(ForwardRef(Link))
    className="title"
    href="/items/1"
  >
    <WithStyles(ForwardRef(Typography))
      gutterBottom={true}
      variant="body2"
    >
      iphone
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Link))>,
]
`;
